src/Screens/Main/Settings/UserAccount/MentorForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import RN from 'react-native';
+import * as redux from 'redux';
+import { useDispatch, useSelector } from 'react-redux';
+
+import * as config from '../../../../api/config';
+import * as actions from '../../../../state/actions';
+import * as mentorState from '../../../../state/reducers/mentors';
+
+import Button from '../../../components/Button';
+import Message from '../../../components/Message';
+import ToggleSwitch from '../../../components/ToggleSwitch';
+import colors from '../../../components/colors';
+import fonts from '../../../components/fonts';
+
+type Props = {
+  userId: string;
+};
+
+export default ({ userId }: Props) => {
+  const mentor = useSelector(mentorState.getMentorByUserId(userId));
+
+  const dispatch = useDispatch<redux.Dispatch<actions.Action>>();
+
+  const openProfile = () => {
+    RN.Linking.openURL(config.loginUrl);
+  };
+
+  const changeVacationStatus = (): boolean => {
+    if (typeof mentor !== 'undefined') {
+      dispatch({
+        type: 'mentor/changeVacationStatus/start',
+        payload: { mentor },
+      });
+
+      return true;
+    }
+
+    return false;
+  };
```

Comment:
When remove the local state of the Switch component, this does not have to return anything.